ts/screens/wallet/TransactionDetailsScreen.tsx

```diff
@@ -154,12 +143,21 @@ export class TransactionDetailsScreen extends React.Component<Props, State> {
         headerContents={this.getSubHeader()}
         cardType={{ type: CardEnum.HEADER, card: this.props.selectedCard }}
         showPayButton={false}
+        allowGoBack={!this.props.navigation.getParam("paymentCompleted", false)}
       >
         <Content scrollEnabled={false} style={WalletStyles.whiteContent}>
           <Grid>
             <Row style={styles.titleRow}>
               <H3>{I18n.t("wallet.transactionDetails")}</H3>
-              <IconFont name="io-close" size={variables.iconSizeBase} />
+              <IconFont
+                name="io-close"
+                size={variables.iconSizeBase}
+                onPress={() =>
+                  this.props.navigation.getParam("paymentCompleted", false)
```

Comment:
again, set `const isPaymentCompleted = this.props.navigation.getParam("paymentCompleted", false)` before rendering the component (i.e. line 138)